### PR TITLE
feat(test/e2e): add boot and archive nodes

### PR DESCRIPTION
**Pull request by Lazar955**
Modifies geth template for archive mode or seed mode. Utilizes cmtbfts node mode for starting archive/seed nodes. 

task: none

### CODE REVIEW

File: test/e2e/types/testnet.go [hunk 1]
```diff
@@ -16,6 +17,13 @@ import (
 	"github.com/ethereum/go-ethereum/p2p/enode"
 )
 
+type GcMode string
```
Comment (corverroos) [inline test/e2e/types/testnet.go:20]: add comment please

File: test/e2e/types/testnet.go [hunk 2]
```diff
@@ -80,6 +90,15 @@ func (o OmniEVM) BootNodesStr() string {
 	return strings.Join(resp, ",")
 }
 
+func (o OmniEVM) BootNodesStrArr() string {
```
Comment (corverroos) [inline test/e2e/types/testnet.go:93]: add a comment explain the different between this and above please, also add comment above please

File: test/e2e/vmcompose/testdata/TestSetup_127_0_0_1_compose.yaml.golden [hunk 3]
```diff
@@ -48,6 +48,7 @@ services:
     container_name: validator01_evm
     image: "ethereum/client-go:latest"
     command:
+      - --config=/geth/config.toml
```
Comment (corverroos) [inline test/e2e/vmcompose/testdata/TestSetup_127_0_0_1_compose.yaml.golden:51]: not sure we should do command line and file based, I think we should pick one

File: test/e2e/app/template/geth.toml.tmpl [hunk 4]
```diff
@@ -0,0 +1,100 @@
+[Eth]
```
Comment (corverroos) [inline test/e2e/app/template/geth.toml.tmpl:1]: not sure a dedicated folder is required, list just put this in `app/`

File: test/e2e/app/setup.go [hunk 5]
```diff
@@ -431,3 +460,26 @@ func logConfig() log.Config {
 		Color:  log.ColorForce,
 	}
 }
+
+//go:embed template/geth.toml.tmpl
+var gethTomlTemplate []byte
+
+// WriteGethConfigTOML writes the toml config to disk.
+func WriteGethConfigTOML(cfg any, path string) error {
```
Comment (corverroos) [inline test/e2e/app/setup.go:468]: please use proper types, not `any`

File: test/e2e/app/setup.go [hunk 6]
```diff
@@ -105,7 +108,21 @@ func Setup(ctx context.Context, def Definition, agentSecrets agent.Secrets, test
...
+		case e2e.ModeValidator, e2e.ModeFull:
+			pruningOption = "nothing"
+			minRetainBlocks = 0
+		case e2e.ModeSeed, e2e.ModeLight:
+			pruningOption = "everything"
+			minRetainBlocks = 1
+		default:
+			pruningOption = "nothing"
+			minRetainBlocks = 0
+		}
```
Comment (corverroos) [inline test/e2e/app/setup.go:123]: do this inside `writeHaloConfig`

File: test/e2e/app/rpc.go [hunk 7]
```diff
@@ -152,3 +157,45 @@ func waitForAllNodes(ctx context.Context, testnet *e2e.Testnet, height int64, ti
 
 	return lastHeight, nil
 }
+
+// waitForGethPeers waits for all geth nodes to connect to each other.
+func waitForGethPeers(ctx context.Context, testnet *e2etypes.Testnet, wanted int64) error {
```
Comment (corverroos) [inline test/e2e/app/rpc.go:162]: why is this required?

File: test/e2e/app/template/geth.toml.tmpl [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+[Node.HTTPTimeouts]
+ReadTimeout = 30000000000
+ReadHeaderTimeout = 30000000000
+WriteTimeout = 30000000000
+IdleTimeout = 120000000000
+
+[Metrics]
+HTTP = "127.0.0.1"
+Port = 6060
+InfluxDBEndpoint = "http://localhost:8086"
```
Comment (corverroos) [inline test/e2e/app/template/geth.toml.tmpl:93]: we don't have this

File: test/e2e/app/template/geth.toml.tmpl [hunk 9]
```diff
@@ -0,0 +1,100 @@
+[Eth]
+NetworkId = 0
+SyncMode = "full"
+EthDiscoveryURLs = []
```
Comment (corverroos) [inline test/e2e/app/template/geth.toml.tmpl:4]: can we remove everything that:
 - is default
 - we don't use

File: test/e2e/vmcompose/testdata/TestSetup_127_0_0_1_compose.yaml.golden (hunk 3 above)
Comment (Lazar955) [reply test/e2e/vmcompose/testdata/TestSetup_127_0_0_1_compose.yaml.golden:51]: Makes sense, will move everything in file, as staticNodes can't be passed as cmd arg

File: test/e2e/manifests/devnet1.toml [hunk 10]
```diff
@@ -8,3 +8,9 @@ prometheus = true
 
 [node.validator01]
 [node.validator02]
+
+[node.seed01]
+mode = "seed"
+
+#[node.fullnode01]
```
[the remark this answers is not in the document]
Comment (Lazar955) [reply test/e2e/manifests/devnet1.toml:15]: yeah will remove, this should go into staging manifest, was testing with devnet

File: test/e2e/vmcompose/testdata/TestSetup_127_0_0_1_compose.yaml.golden (hunk 3 above)
Comment (Lazar955) [reply test/e2e/vmcompose/testdata/TestSetup_127_0_0_1_compose.yaml.golden:51]: Actually, we have to have a split between config file and args

File: e2e/app/setup.go [hunk 11]
```diff
@@ -332,10 +335,24 @@ func MakeConfig(node *e2e.Node, nodeDir string) (*config.Config, error) {
...
+
+	switch mode {
+	case e2e.ModeValidator, e2e.ModeFull:
+		cfg.PruningOption = "nothing"
+		cfg.MinRetainBlocks = 0
+	case e2e.ModeSeed, e2e.ModeLight:
+		cfg.PruningOption = "everything"
+		cfg.MinRetainBlocks = 1
+	default:
+		cfg.PruningOption = "nothing"
```
Comment (corverroos) [inline e2e/app/setup.go:349]: lets de "Default" here.

**Review by corverroos — APPROVED** [review]

**Review by pavelnikolov — APPROVED** [review]

File: e2e/app/run.go [hunk 12]
```diff
@@ -69,6 +70,10 @@ func Deploy(ctx context.Context, def Definition, cfg DeployConfig) (types.Deploy
 			return nil, nil, err
 		}
 
+		// todo(lazar): remove this when we figure out why txs are stuck in geth mempool upon initial run
+		// wait for geth nodes to connect
+		time.Sleep(3 * time.Second)
```
[the remark this answers is not in the document]
Comment (Lazar955) [reply e2e/app/run.go]: yep removed

File: e2e/app/setup.go [hunk 13]
```diff
@@ -332,10 +335,24 @@ func MakeConfig(node *e2e.Node, nodeDir string) (*config.Config, error) {
...
+		cfg.MinRetainBlocks = 0
```
[the remark this answers is not in the document]
Comment (Lazar955) [reply e2e/app/setup.go:350]: Unlikely to be a new mode, open to suggestions for default values

[the remark this answers is not in the document]
Comment (Lazar955) [reply e2e/app/setup.go:350]: changed to use the "default" pruning
